New version of Flake8 tool finds new violations

./BasicDevTests.py:51:17: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/path_utilities_test.py:581:14: F541 f-string is missing placeholders
./edk2toollib/uefi/edk2/build_objects/dsc_translator.py:27:13: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/dsc_parser.py:247:17: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/base_parser.py:117:13: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/base_parser.py:601:28: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/base_parser.py:615:25: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/base_parser.py:710:28: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/base_parser.py:721:31: E741 ambiguous variable name 'l'
./edk2toollib/uefi/edk2/parsers/dsc_parser_test.py:62:22: F541 f-string is missing placeholders
./edk2toollib/windows/locate_tools.py:61:27: F541 f-string is missing placeholders

--- a/BasicDevTests.py
+++ b/BasicDevTests.py
@@ -43,13 +43,13 @@ def TestNoSpaces(apath):
 
 
 def TestRequiredLicense(apath):
-    lic = ["SPDX-License-Identifier: BSD-2-Clause-Patent"]
+    licenses = ["SPDX-License-Identifier: BSD-2-Clause-Patent", ]
     try:
         with open(apath, "rb") as fobj:
             contents = fobj.read().decode()
             found = False
-            for l in lic:
-                if l in contents:
+            for lic in licenses:
+                if lic in contents:
                     found = True
                     break
             if not found:

--- a/edk2toollib/uefi/edk2/build_objects/dsc_translator.py
+++ b/edk2toollib/uefi/edk2/build_objects/dsc_translator.py
@@ -24,8 +24,8 @@ class DscTranslator():
         file_path = os.path.abspath(filepath)
         f = open(file_path, "w")
         lines = cls._GetDscLinesFromDscObj(dsc_obj)
-        for l in lines:
-            f.write(l + "\n")
+        for line in lines:
+            f.write(line + "\n")
         f.close()
 
     @classmethod

--- a/edk2toollib/uefi/edk2/parsers/base_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/base_parser.py
@@ -376,13 +376,13 @@ class BaseParser(object):
         if not text.lower().startswith("!if "):
             raise RuntimeError(f"Invalid conditional cannot be validated: {text}")
         text = text[3:].strip()
-        self.Logger.debug(f"STAGE 1: {text}")
+        logging.debug(f"STAGE 1: {text}")
         text = self.ReplaceVariables(text)
-        self.Logger.debug(f"STAGE 2: {text}")
+        logging.debug(f"STAGE 2: {text}")
         tokens = self._TokenizeConditional(text)
-        self.Logger.debug(f"STAGE 3: {tokens}")
+        logging.debug(f"STAGE 3: {tokens}")
         expression = self._ConvertTokensToPostFix(tokens)
-        self.Logger.debug(f"STAGE 4: {expression}")
+        logging.debug(f"STAGE 4: {expression}")
 
         # Now we evaluate the post fix expression
         if len(expression) == 0:
@@ -432,7 +432,7 @@ class BaseParser(object):
                 expression = new_expression
 
         final = self.ConvertToInt(expression[0])
-        self.Logger.debug(f" FINAL {expression} {final}")
+        logging.debug(f" FINAL {expression} {final}")
 
         return bool(final)
 

--- a/edk2toollib/uefi/edk2/parsers/base_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/base_parser.py
@@ -113,10 +113,10 @@ class BaseParser(object):
 
         """
         self.Logger.debug("Writing all lines to file: %s" % filepath)
-        f = open(filepath, "w")
-        for l in self.Lines:
-            f.write(l + "\n")
-        f.close()
+        file_handle = open(filepath, "w")
+        for line in self.Lines:
+            file_handle.write(line + "\n")
+        file_handle.close()
     #
     # do logical comparisons
     #
@@ -598,21 +598,21 @@ class BaseParser(object):
 
         return ret
 
-    def IsGuidString(self, l):
+    def IsGuidString(self, line):
         """
         will return true if the the line has
         = { 0xD3B36F2C, 0xD551, 0x11D4, { 0x9A, 0x46, 0x00, 0x90, 0x27, 0x3F, 0xC1, 0x4D }}
         Args:
-          l:
+          line:
 
         Returns:
 
         """
-        if(l.count("{") == 2 and l.count("}") == 2 and l.count(",") == 10 and l.count("=") == 1):
+        if(line.count("{") == 2 and line.count("}") == 2 and line.count(",") == 10 and l.count("=") == 1):
             return True
         return False
 
-    def ParseGuid(self, l):
+    def ParseGuid(self, line):
         """
         parse a guid into a different format
         Will throw exception if missing any of the 11 parts of isn't long enough
@@ -622,9 +622,10 @@ class BaseParser(object):
         Returns: a string of the guid. ex: D3B36F2C-D551-11D4-9A46-0090273FC14D
 
         """
-        entries = l.lstrip(' {').rstrip(' }').split(',')
+        entries = line.lstrip(' {').rstrip(' }').split(',')
         if len(entries) != 11:
-            raise RuntimeError(f"Invalid GUID found {l}. We are missing some parts since we only found: {len(entries)}")
+            raise RuntimeError(
+                f"Invalid GUID found {line}. We are missing some parts since we only found: {len(entries)}")
         gu = entries[0].lstrip(' 0').lstrip('x').strip()
         # pad front until 8 chars
         while(len(gu) < 8):
@@ -707,28 +708,28 @@ class HashFileParser(BaseParser):
     def __init__(self, log):
         BaseParser.__init__(self, log)
 
-    def StripComment(self, l):
+    def StripComment(self, line):
         """
 
         Args:
-          l:
+          line:
 
         Returns:
 
         """
-        return l.split('#')[0].strip()
+        return line.split('#')[0].strip()
 
-    def ParseNewSection(self, l):
+    def ParseNewSection(self, line):
         """
 
         Args:
-          l:
+          line:
 
         Returns:
 
         """
-        if(l.count("[") == 1 and l.count("]") == 1):  # new section
-            section = l.strip().lstrip("[").split(".")[0].split(",")[0].rstrip("]").strip()
-            self.CurrentFullSection = l.strip().lstrip("[").split(",")[0].rstrip("]").strip()
+        if(line.count("[") == 1 and line.count("]") == 1):  # new section
+            section = line.strip().lstrip("[").split(".")[0].split(",")[0].rstrip("]").strip()
+            self.CurrentFullSection = line.strip().lstrip("[").split(",")[0].rstrip("]").strip()
             return (True, section)
         return (False, "")

--- a/edk2toollib/uefi/edk2/parsers/base_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/base_parser.py
@@ -376,13 +376,13 @@ class BaseParser(object):
         if not text.lower().startswith("!if "):
             raise RuntimeError(f"Invalid conditional cannot be validated: {text}")
         text = text[3:].strip()
-        logging.debug(f"STAGE 1: {text}")
+        self.Logger(f"STAGE 1: {text}")
         text = self.ReplaceVariables(text)
-        logging.debug(f"STAGE 2: {text}")
+        self.Logger(f"STAGE 2: {text}")
         tokens = self._TokenizeConditional(text)
-        logging.debug(f"STAGE 3: {tokens}")
+        self.Logger(f"STAGE 3: {tokens}")
         expression = self._ConvertTokensToPostFix(tokens)
-        logging.debug(f"STAGE 4: {expression}")
+        self.Logger(f"STAGE 4: {expression}")
 
         # Now we evaluate the post fix expression
         if len(expression) == 0:
@@ -432,7 +432,7 @@ class BaseParser(object):
                 expression = new_expression
 
         final = self.ConvertToInt(expression[0])
-        logging.debug(f" FINAL {expression} {final}")
+        self.Logger(f" FINAL {expression} {final}")
 
         return bool(final)
 

--- a/edk2toollib/uefi/edk2/parsers/base_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/base_parser.py
@@ -608,7 +608,7 @@ class BaseParser(object):
         Returns:
 
         """
-        if(line.count("{") == 2 and line.count("}") == 2 and line.count(",") == 10 and l.count("=") == 1):
+        if(line.count("{") == 2 and line.count("}") == 2 and line.count(",") == 10 and line.count("=") == 1):
             return True
         return False
 

--- a/edk2toollib/uefi/edk2/parsers/base_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/base_parser.py
@@ -13,7 +13,7 @@ class BaseParser(object):
     """ """
     operators = ["OR", "AND", "IN", "==", "!=", ">", "<", "<=", ">="]
 
-    def __init__(self, log=""):
+    def __init__(self, log="BaseParser"):
         self.Logger = logging.getLogger(log)
         self.Lines = []
         self.LocalVars = {}
@@ -376,13 +376,13 @@ class BaseParser(object):
         if not text.lower().startswith("!if "):
             raise RuntimeError(f"Invalid conditional cannot be validated: {text}")
         text = text[3:].strip()
-        self.Logger(f"STAGE 1: {text}")
+        self.Logger.debug(f"STAGE 1: {text}")
         text = self.ReplaceVariables(text)
-        self.Logger(f"STAGE 2: {text}")
+        self.Logger.debug(f"STAGE 2: {text}")
         tokens = self._TokenizeConditional(text)
-        self.Logger(f"STAGE 3: {tokens}")
+        self.Logger.debug(f"STAGE 3: {tokens}")
         expression = self._ConvertTokensToPostFix(tokens)
-        self.Logger(f"STAGE 4: {expression}")
+        self.Logger.debug(f"STAGE 4: {expression}")
 
         # Now we evaluate the post fix expression
         if len(expression) == 0:
@@ -432,7 +432,7 @@ class BaseParser(object):
                 expression = new_expression
 
         final = self.ConvertToInt(expression[0])
-        self.Logger(f" FINAL {expression} {final}")
+        self.Logger.debug(f" FINAL {expression} {final}")
 
         return bool(final)
 

--- a/edk2toollib/uefi/edk2/parsers/dsc_parser.py
+++ b/edk2toollib/uefi/edk2/parsers/dsc_parser.py
@@ -244,8 +244,8 @@ class DscParser(HashFileParser):
 
     def __ProcessDefines(self, lines):
         if(len(lines) > 0):
-            for l in lines:
-                (line, add) = self.__ParseDefineLine(l)
+            for raw_line in lines:
+                (line, add) = self.__ParseDefineLine(raw_line)
                 self.__ProcessDefines(add)
 
     def ResetParserState(self):

--- a/edk2toollib/uefi/edk2/parsers/dsc_parser_test.py
+++ b/edk2toollib/uefi/edk2/parsers/dsc_parser_test.py
@@ -59,7 +59,7 @@ class TestDscParserIncludes(unittest.TestCase):
         file1_name = "file1.dsc"
         file1_path = os.path.join(workspace, file1_name)
 
-        file1_data = f"!include BAD_FILE.dsc"
+        file1_data = "!include BAD_FILE.dsc"
 
         TestDscParserIncludes.write_to_file(file1_path, file1_data)
 

--- a/edk2toollib/uefi/edk2/path_utilities_test.py
+++ b/edk2toollib/uefi/edk2/path_utilities_test.py
@@ -578,7 +578,7 @@ class PathUtilitiesTest(unittest.TestCase):
         self.assertEqual(pathobj.GetAbsolutePathOnThisSytemFromEdk2RelativePath(rp), ep)
 
         # file not in workspace
-        rp = f"DoesNotExistPkg/module2/X64/TestFile.c"
+        rp = "DoesNotExistPkg/module2/X64/TestFile.c"
         self.assertIsNone(pathobj.GetAbsolutePathOnThisSytemFromEdk2RelativePath(rp))
 
         # pass in bad parameter

--- a/edk2toollib/windows/locate_tools.py
+++ b/edk2toollib/windows/locate_tools.py
@@ -58,7 +58,7 @@ def _DownloadVsWhere(unpack_folder: os.PathLike = None):
             with urllib.request.urlopen(__URL) as response, open(out_file_name, 'wb') as out_file:
                 out_file.write(response.read())
         except urllib.error.HTTPError as e:
-            logging.error(f"We ran into an issue when getting VsWhere")
+            logging.error("We ran into an issue when getting VsWhere")
             raise e
 
     # do the hash to make sure the file is good

